network/peers/connections/conn_gater.go

```diff
@@ -89,11 +91,15 @@ func (n *connGater) InterceptAccept(multiaddrs libp2pnetwork.ConnMultiaddrs) boo
 // InterceptSecured is called for both inbound and outbound connections,
 // after a security handshake has taken place and we've authenticated the peer.
 func (n *connGater) InterceptSecured(direction libp2pnetwork.Direction, id peer.ID, multiaddrs libp2pnetwork.ConnMultiaddrs) bool {
-	if n.isBadPeer(n.logger, id) {
-		n.logger.Debug("rejecting inbound connection due to bad peer", fields.PeerID(id))
+	if direction == libp2pnetwork.DirUnknown {
 		return false
 	}
-	return true
+
+	if direction == libp2pnetwork.DirOutbound {
+		return n.isBadOutbound(n.logger, id)
+	} else {
+		return n.isBadInbound(n.logger, id)
+	}
```

Comment:
And here we essentially do `isBadOutbound` check 2nd time (because the first time, I think, we did it in `InterceptAddrDial`), which maybe means - instead of all this added code in `InterceptSecured` we maybe should just check `isBadInbound` in `InterceptAccept ` ?